README.md

```diff
@@ -1,109 +1,56 @@
-    url  "https://jitpack.io"
-  }
+  maven { url  "https://jitpack.io" }
 }
 
 dependencies {
-  compile "com.github.Discord4J.Discord4J:Discord4J-core:@VERSION@"
+  implementation "com.discord4j.discord4j:discord4j-core:@VERSION@"
 }
 ```
-### With SBT
-In your `build.sbt` add (without the ellipses):
-```sbt
-libraryDependencies ++= Seq(
-  "com.github.Discord4J.Discord4J" % "Discord4J-core" % "@VERSION@"
-)
 
-resolvers += "jitpack.io" at "https://jitpack.io"
-```
-### Manually with the shaded jar
-TODO: Add shaded jar links for core
-If you don't use Maven nor Gradle (which you really should, because it's a lot more flexible and allows you to update easily), you can always [grab the shaded jar file](http://discord4j.com/Discord4J/downloads.html) (which has all the D4J dependencies inside), and link it in your IntelliJ or Eclipse project.
-#### IntelliJ
-Module Settings > Dependencies > click the + > JARs or directories > Select your JAR file
-#### Eclipse
-Project Properties > Java Build Path > Add the jar file
+## Quick Example
+```java
+final DiscordClient client = new ClientBuilder("token").build();
 
-## So, how do I use this?
-### Tutorials/Resources
-* TODO
+client.getEventDispatcher().on(ReadyEvent.class)
+        .subscribe(ready -> System.out.println("Logged in as " + ready.getSelf().getUsername()));
 
-### Starting with the API
-TODO
+client.getEventDispatcher().on(MessageCreateEvent.class)
+        .map(MessageCreateEvent::getMessage)
+        .filter(msg -> msg.getContent().map(content -> content.equals("!ping")).orElse(false))
+        .flatMap(Message::getChannel)
+        .flatMap(channel -> channel.createMessage(spec -> spec.setContent("Pong!")))
+        .subscribe();
 
-### More examples
-TODO
-
-## Projects using Discord4J
-TODO
+client.login().block();
+```
 
-## Deprecation policy
-Due to the nature of the Discord API, in general any deprecations found in the API should not be expected to last past the current
-version. Meaning that if a method is deprecated on version 2.1.0, do not assume the method will be available in version 2.2.0.
+## Modules
+Discord4J is highly oriented towards customizability and fine-grained control. To achieve this, the project is divided into several "modules" which can be used separately depending on your use case.
 
-In some extraordinary cases, we may be forced to remove APIs without going through a deprecation cycle.
+### [Core](./core/README.md)
+The `core` module combines the other modules to form high-level abstractions for the entire Discord Bot API. This is the module most users will want when making bots.
 
-## Development
-The Discord API is still in development. Functions may break at any time.  
-In such an event, please contact me or submit a pull request.
+### [Rest](./rest/README.md)
+The `rest` module provides a low-level HTTP client specifically for Discord which properly handles Discord's [ratelimiting system](https://discordapp.com/developers/docs/topics/rate-limits).
```

Comment:
If the gateway module links to gateway api docs, this should link to rest docs. Also REST should be all upper case

Reply:
Such a section does not exist.

Reply:
Hm I for some reason thought there was a general rest-ish section, but i guess not